tests/test_ui_comprehensive.py

```
@@ -0,0 +1,204 @@
+                    callbacks["Hex"] = cb
+                return m
+            m.on_value_change.side_effect = on_val_change
+            return m
+
+        mock_ui.input = mock_element
+        mock_ui.number = mock_element
+        mock_ui.switch = mock_element
+        mock_ui.select = mock_element
+
+        # Test 1: Normal hex toggles
+        setattr(ui_inst, f'_is_hex_int_val_root', True)
```

Ruff (0.15.15)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)